src/Interpreters/HashJoin.cpp

```diff
@@ -1825,7 +1828,7 @@ void HashJoin::checkTypesOfKeys(const Block & block) const
     }
 }
 
-void HashJoin::joinBlock(Block & block, ExtraBlockPtr & not_processed)
```

Comment:
Can't we leave this version as is and implement both `joinBlock` and `joinBlockWithStreamOutput` with minimal changes ? Then we can use regular version where profit from streaming is not used (ConcurrentHashJoin, GraceHash).